sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -163,91 +196,131 @@ public void addAssignedOverlayPort(long port) {
-                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build()); // Set later by ExecutorRequirement
-        // Populate ContainerInfo with the appropriate information from PodSpec.
-        // This includes networks, rlimits, secret volumes...
-        Protos.ContainerInfo containerInfo = getContainerInfo(podSpec);
-        if (containerInfo != null) {
-            executorInfoBuilder.setContainer(containerInfo);
-        }
-
-        // command and user:
-        Protos.CommandInfo.Builder executorCommandBuilder = executorInfoBuilder.getCommandBuilder().setValue(
-                "export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH && " +
-                        "export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so) && " +
-                        "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && " +
-                        // Remove Xms/Xmx if +UseCGroupMemoryLimitForHeap or equivalent detects cgroups memory limit
-                        "export JAVA_OPTS=\"-Xms128M -Xmx128M\" && " +
-                        "$MESOS_SANDBOX/executor/bin/executor");
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build());
+        Protos.ContainerInfo.Builder containerBuilder = getContainerInfo(podSpec, true, false).toBuilder();
 
-        if (podSpec.getUser().isPresent()) {
-            executorCommandBuilder.setUser(podSpec.getUser().get());
-        }
+        executorInfoBuilder.getLabelsBuilder().addLabelsBuilder()
+                .setKey("DCOS_SPACE")
+                .setValue(getDcosSpaceLabel());
 
-        // Required URIs from the scheduler environment:
-        executorCommandBuilder.addUrisBuilder().setValue(schedulerFlags.getLibmesosURI());
-        executorCommandBuilder.addUrisBuilder().setValue(schedulerFlags.getJavaURI());
+        if (useDefaultExecutor) {
+            executorInfoBuilder.setType(Protos.ExecutorInfo.Type.DEFAULT)
+                    .setFrameworkId(frameworkID);
+            executorInfoBuilder.getContainerBuilder()
+                    .setType(Protos.ContainerInfo.Type.MESOS)
+                    .getMesosBuilder().clearImage();
+
+            if (!podInstance.getPod().getNetworks().isEmpty()) {
+                containerBuilder.addAllNetworkInfos(
+                        podSpec.getNetworks().stream()
+                                .map(PodInfoBuilder::getNetworkInfo).collect(Collectors.toList()));
+            }
+        } else {
+            // command and user:
+            Protos.CommandInfo.Builder executorCommandBuilder = executorInfoBuilder.getCommandBuilder().setValue(
+                    "export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH && " +
+                    "export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so) && " +
+                    "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && " +
```

Comment:
(not a thing to fix) here's an example of dynamically setting `JAVA_HOME` within the `cmd`